### ISSUE TITLE
[b17] Completely clogged Flux Condenser array causes crash

### ISSUE THREAD

**Issue by Reassembly** [Azanor/thaumcraft-beta#1151]
**Crash log:** https://paste.dimdev.org/ipozeciwek.mccrash

**Apparent issue:** Periodically, an active Flux Condenser will select one unclogged Array element and generate a "sparking" visual effect on it, calling Random.nextInt(int bound) with the count of unclogged elements, and using that as an index into a list of unclogged elements. If a Condenser is active but has no unclogged elements attached (e.g. all elements are clogged), then this routine calls Random.nextInt with a bound of 0. The following code implies that the expected behaviour is that this will merely return a null reference, but instead it triggers an IllegalArgumentException. Random.nextInt() requires a positive bound, if a bound is specified, and will trigger an exception if called with a zero or negative bound.

**Replication note:** Oddly, this issue did not seem to occur while the game was running and array elements became clogged - the crash occurred only after shutting down and reloading a world which contained Condensers with completely-clogged arrays. It is possible that the list of unclogged elements was not being updated correctly _while_ the world was running, and was only correctly refreshed to a zero-element list once the world was shut down and reloaded. When replicating this issue, it might be necessary to completely clog a Condenser array (even a single-element array will do), _then_ shut down and reload the world.

EDIT: Duplicate of #1095, derp.
